engine/src/main/java/org/terasology/world/zones/LayeredZoneRegionFunction.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.world.zones;
+
+import org.terasology.math.geom.BaseVector3i;
+import org.terasology.math.geom.Vector2i;
+import org.terasology.module.sandbox.API;
+import org.terasology.world.chunks.ChunkConstants;
+import org.terasology.world.generation.Region;
+import org.terasology.world.generation.facets.SurfaceHeightFacet;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * A function that can be used as a {@link Zone#regionFunction} to create zones that are layered on top of each other.
+ *
+ * These layers are ordered according to {@link #ordering}, and have a width of {@link #minWidth}.
```

Comment:
Yeah, I wasn't sure either. I think that depth might imply how far away from the surface it is, rather than how thick it is. Extent is closer, but that makes me think of the whole extent (considering all 3 dimensions), rather than just how wide it is. I think that width/height both have the same problem of referring to different things depending on where your axes are placed. I think thickness is the closest word I've found, and it seems like it fits nicely.